client/scripts/views/components/admin_panel.ts

```diff
@@ -0,0 +1,634 @@
+                size: 'xs',
+                style: 'padding-left: 2px;',
+                onclick: () => {
+                  $.post(`${app.serverUrl()}/upgradeMember`, {
+                    ...chainOrCommObj,
+                    new_role: 'member',
+                    address: role.Address.address,
+                    jwt: app.login.jwt,
+                  }).then((res) => {
+                    if (res.status !== 'Success') {
+                      throw new Error(`got unsuccessful status: ${res.status}`);
+                    }
+                    vnode.attrs.onRoleUpdate(role, res.result);
+                  }).catch((e) => console.error('Failed To demote admin'));
+                },
+              }),
+            ]);
+          })
+      ]);
+  }
+};
+
+interface ICommunityMetadataState {
+    name: string;
+    description: string;
+    url: string;
+    privacyToggled: boolean;
+    invitesToggled: boolean;
+}
+
+interface IChainCommunityAttrs {
+  community?: CommunityInfo;
+  chain?: ChainInfo;
+  onChangeHandler: Function;
+  onRoleUpdate: Function;
+  admins;
+  mods;
+}
+
+interface ITableRowAttrs {
+  title: string;
+  defaultValue: string;
+  disabled?: boolean;
+  onChangeHandler: Function;
+}
+
+const TableRow: m.Component<ITableRowAttrs> = {
+  view: (vnode) => {
+    return m('tr', [
+      m('td', { style: 'width: 100px' }, vnode.attrs.title),
+      m('td', [
+        m(Input, {
+          defaultValue: vnode.attrs.defaultValue,
+          fluid: true,
+          disabled: vnode.attrs.disabled || false,
+          onkeyup: (e) => { vnode.attrs.onChangeHandler((e.target as any).value); },
+        }),
+      ]),
+    ]);
+  }
+};
+
+interface IToggleRowAttrs {
+  title: string;
+  defaultValue: boolean;
+  disabled?: boolean;
+  onToggle: Function;
+}
+
+const ToggleRow: m.Component<IToggleRowAttrs, {toggled: boolean, checked: boolean}> = {
+  oninit: (vnode) => {
+    vnode.state.toggled = false;
+    vnode.state.checked = vnode.attrs.defaultValue;
+  },
+  view: (vnode) => {
+    return m('tr', [
+      m('td', vnode.attrs.title),
+      m('td', [
+        m(Switch, {
+          checked: vnode.state.checked,
+          disabled: vnode.attrs.disabled || false,
+          onchange: () => {
+            vnode.state.toggled = !vnode.state.toggled;
+            vnode.state.checked = !vnode.state.checked;
+            vnode.attrs.onToggle(vnode.state.toggled);
+          },
+          // style: 'margin-left: 20px;',
+        })
+      ])
+    ]);
+  },
+};
+
+const CommunityMetadata: m.Component<IChainCommunityAttrs, ICommunityMetadataState> = {
+  oninit: (vnode) => {
+    vnode.state.name = vnode.attrs.community.name;
+    vnode.state.description = vnode.attrs.community.description;
+    vnode.state.url = vnode.attrs.community.id;
+  },
+  view: (vnode) => {
+    return m('CommunityMetadata', [m(Table, {
+      bordered: false,
+      interactive: false,
+      striped: false,
+      class: '.community.metadata',
```

Comment:
- The class property is not a selector, so it should be 'community metadata' to get two classes and 'community-metadata' for one class.
- We can just specify the class in the component's name. See above
- We should avoid using a class name like `metadata` because it will cause conflicts.

Reply:
Regarding your last point, even if the `metadata` class is encapsulated by the `.AdminPanel` class in the scss? I'll make it more specific, though, for clarity.